components/mdi/mdi_can_interface/Core/Src/can.c

```diff
@@ -0,0 +1,126 @@
+/**
+ * Functions used in the main file of the MDI firmware
+ */
+#include "main.h"
+#include "can.h"
+#include "stm32f1xx_hal_can.h"
+#include "stm32f1xx_hal.h"
+#include "stm32f1xx.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+
+/**
+ * @brief Takes data from pedal (last 32 bits of message) and converts to 10 bit value. 
+ * @param pedal_data: bits 32-63 of 0x401 message data element 
+ * @retval parsed_voltage is the 10 bits to be sent to the DAC
+ * 
```

Comment:
Couple comment formatting things:

`@brief` can be omitted before the first line of a doxygen comment and this makes things a little cleaner. Regardless, putting a new line after the brief description statement is standard.

Don't put a colon after a parameter name

`@returns` is a bit more natural than `@retval` if you want to use that instead

Nit: remove extra newline at the end

```suggestion
 * Takes data from pedal (last 32 bits of message) and converts to 10 bit value.
 *
 * @param pedal_data bits 32-63 of 0x401 message data element 
 * @returns parsed_voltage the 10 bits to be sent to the DAC
```